lib/index.coffee

```diff
@@ -49,21 +49,11 @@ exports.create = (string, options, callback) ->
       return callback(error)
 
     file = Buffer.concat(stdout)
-    isFileBuffer = /^\%PDF/.test(file.slice(0, 4).toString())
-
-    if options.filename
-      callback(null, file.toString())
-
-    else if !isFileBuffer
-      filename = file.toString()
-      fs.readFile filename, (err, buffer) ->
-        return callback(err) if err
-        fs.unlink filename, (err) ->
-          callback(err, buffer)
-
-    else
-      callback(null, file)
 
+    try
+      file = JSON.parse(file)
+    catch err
```

Comment:
That's a wrong behaviour. We should return the error in the callback if there's one.